lib/HLSL/DxilRootSignature.cpp

```diff
@@ -146,7 +146,7 @@ SimpleSerializer::~SimpleSerializer() {
     m_pSegment = pSegment->pNext;
 
     if (pSegment->bOwner) {
-      delete pSegment->pData;
+      delete[] (char*)pSegment->pData;
```

Comment:
why is this necessary? `delete` on a `void*` is valid, is it not?

Reply:
In addition to the caveats above, in this case, the allocation was done as an array. There's a good chance the compiler does the right thing with PODs, but using `delete` instead of `delete[]` is clearly a violation of the letter of the law.

Reply:
Yes. Agreed.